src/main/java/uk/gov/hmcts/reform/et/syaapi/service/CaseDocumentService.java

```diff
@@ -0,0 +1,211 @@
+     * @param authTokenGenerator the {@link AuthTokenGenerator} used to generate tokens for communicating with the
+     *                           Case Document API
+     * @param caseDocApiUrl      the URL to call the Case Document API
+     */
+    public CaseDocumentService(RestTemplate restTemplate,
+                               AuthTokenGenerator authTokenGenerator,
+                               @Value("${case_document_am.url}/cases/documents}")
+                                   String caseDocApiUrl) {
+        this.restTemplate = restTemplate;
+        this.authTokenGenerator = authTokenGenerator;
+        this.caseDocApiUrl = caseDocApiUrl;
+    }
+
+    /**
+     * Given a file to upload, this will upload the file to the CCD document API and give back a unique ULR to access
+     * the uploaded file.
+     *
+     * @param authToken  the caller's bearer token used to verify the caller
+     * @param caseTypeId the area the file belongs to e.g. ET_EnglandWales
+     * @param file       the file to be uploaded
+     * @return the URL of the document we have just uploaded
+     * @throws CaseDocumentException if a problem occurs whilst uploading the document via API
+     */
+    public URI uploadDocument(String authToken, String caseTypeId, MultipartFile file) throws CaseDocumentException {
+        DocumentUploadResponse response = attemptWithRetriesToUploadDocumentToCaseDocumentApi(
+            0, authToken, caseTypeId, file);
+
+        CaseDocument caseDocument = validateResponse(
+            Objects.requireNonNull(response), file.getOriginalFilename());
+
+        return getUriFromFile(caseDocument);
+    }
+
+    private URI getUriFromFile(CaseDocument caseDocument) {
+        if (caseDocument.getLinks() != null
+            && caseDocument.getLinks().get("self") != null
+            && caseDocument.getLinks().get("self").get("href") != null) {
+            return URI.create(caseDocument.getLinks().get("self").get("href"));
+        }
+        return URI.create("");
+    }
+
+    private DocumentUploadResponse attemptWithRetriesToUploadDocumentToCaseDocumentApi(int attempts,
+                                                                                      String authToken,
+                                                                                      String caseTypeId,
+                                                                                      MultipartFile file)
+        throws CaseDocumentException {
+        try {
+            return uploadDocumentToCaseDocumentApi(authToken, caseTypeId, file).getBody();
+        } catch (IOException | RestClientException e) {
+            if (attempts < MAX_API_RETRIES) {
+                return attemptWithRetriesToUploadDocumentToCaseDocumentApi(
+                    attempts + 1, authToken, caseTypeId, file);
+            }
+            throw new CaseDocumentException("Failed to upload Case Document", e);
+        }
+    }
```

Comment:
What happened to your use of @ Retryable @DarylBoam ?

Reply:
I wasn't able to trigger the @ Retryable during testing, would need to be called from a new class apparently. Was easier to just use same implementation from ACAS

Reply:
You wet my appetite with @Retryable.  Let's see if we can get this working :)